### PR TITLE
support 'tidb_txn_mode' session variable.

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
The old variable `tidb_pessimistic_lock` can only set the transaction to pessimistic mode.
The new variable `tidb_txn_mode` can set transaction mode to either pessimistic or optimistic.
Which is useful when the config file sets pessimistic mode as default.

### What is changed and how it works?
- Remove `tidb_pessimistic_lock` variable.
- Add `tidb_txn_mode` variable.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by coocood** [comment]
@jackysp @tiancaiamao PTAL

File: session/session.go [hunk 1]
```diff
@@ -1740,8 +1740,14 @@ func (s *session) PrepareTxnCtx(ctx context.Context) {
 	}
 	if !s.sessionVars.IsAutocommit() {
 		txnConf := config.GetGlobalConfig().PessimisticTxn
```
Comment (jackysp) [inline session/session.go]: Rename `txnConf` to `pessimisticTxnConf`?

**Comment by coocood** [comment]
@jackysp PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**